### PR TITLE
Add initial teams tab

### PR DESCRIPTION
**Pull request by akalin-keybase**
For now, just have "create team" and "join team" buttons (that do nothing),
the blurb on the bottom, and a "Read the doc" link.

### CODE REVIEW

File: shared/teams/render.js.flow [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// @flow
```
Comment (akalin-keybase) [inline shared/teams/render.js.flow]: We'll eventually have a teams dialog on mobile right? So I should keep the types into their own .flow file even though it's desktop-only now.

File: shared/teams/index.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+// @flow
+import TeamsContainer from './render'
+import pausableConnect from '../util/pausable-connect'
+import openURL from '../util/open-url'
+
+import type {RouteProps} from '../route-tree/render-route'
+
+// $FlowIssue type this connector
+export default pausableConnect(
```
Comment (akalin-keybase) [inline shared/teams/index.js]: I copied the settings tab code here. Is there a better example I should be using?

File: shared/teams/render.js.flow (hunk 1 above)
Comment (chrisnojima) [reply shared/teams/render.js.flow]: We should likely start with unified views and split them if we need to. You can peek at the mobile designs and see if their different

File: shared/app/tab-bar/index.render.desktop.js [hunk 3]
```diff
@@ -16,6 +16,8 @@ const _icons = {
   [Tabs.profileTab]: {selected: 'icon-nav-people-selected-32', unselected: 'icon-nav-people-32'},
   [Tabs.searchTab]: {selected: 'icon-nav-people-selected-32', unselected: 'icon-nav-people-32'},
   [Tabs.settingsTab]: {selected: 'icon-nav-settings-selected-32', unselected: 'icon-nav-settings-32'},
+  // TODO: Use teams icon when it becomes available.
+  [Tabs.teamsTab]: {selected: 'icon-nav-chat-selected-32', unselected: 'icon-nav-chat-32'},
```
Comment (chrisnojima) [inline shared/app/tab-bar/index.render.desktop.js]: these exist today: https://zpl.io/pbL9pM2

File: shared/teams/index.js (hunk 2 above)
Comment (chrisnojima) [reply shared/teams/index.js]: I'd split these out into name functions. we use mapStateToProps, mapDispatchToProps

File: shared/teams/render.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
+// @flow
+import * as React from 'react'
+import {ClickableBox, Box, Icon, ScrollView, Text} from '../common-adapters'
+import {globalColors, globalStyles} from '../styles'
+
+import type {Props} from './render'
+
+function TeamsRender(props: Props) {
+  // TODO: Find right icon for "Join a team" button.
```
Comment (chrisnojima) [inline shared/teams/render.js]: iconfont-team-join

Comment (chrisnojima) [reply shared/teams/render.js]: small nit we tend to make these arrow functions
```const TeamsRender = (props: Props) => ( <Box```

File: shared/teams/render.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  return (
+    <Box style={{...globalStyles.flexBoxColumn, alignItems: 'center', height: '100%'}}>
+      <Box style={{...globalStyles.flexBoxRow, alignItems: 'center', height: 48}}>
```
Comment (chrisnojima) [inline shared/teams/render.js]: shoud likely use globalMargins when possible

File: shared/teams/render.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+        <ClickableBox onClick={props.onJoinTeam} style={{...globalStyles.flexBoxRow, alignItems: 'center'}}>
+          <Icon type="iconfont-new" style={{color: globalColors.blue}} />
+          <Text type="HeaderLink" style={{padding: 5}}>Join a team</Text>
+        </ClickableBox>
+      </Box>
+      <ScrollView
+        style={{alignSelf: 'stretch'}}
+        contentContainerStyle={{...globalStyles.flexBoxColumn, alignItems: 'center', margin: 10}}
+      >
```
Comment (chrisnojima) [inline shared/teams/render.js]: are you going to have the team rows?

File: shared/teams/render.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+      <Box style={{...globalStyles.flexBoxRow, alignItems: 'center', height: 48}}>
+        <ClickableBox
+          onClick={props.onCreateTeam}
+          style={{...globalStyles.flexBoxRow, alignItems: 'center', marginRight: 40}}
+        >
+          <Icon type="iconfont-new" style={{color: globalColors.blue}} />
+          <Text type="HeaderLink" style={{padding: 5}}>Create a team</Text>
+        </ClickableBox>
+
+        <ClickableBox onClick={props.onJoinTeam} style={{...globalStyles.flexBoxRow, alignItems: 'center'}}>
```
Comment (chrisnojima) [inline shared/teams/render.js]: Its likely worth it to break these out into sub components (in this file)
so you can do
```
const Banner = ({}) => ...
const List = ({}) => ...
const BetaNote = () => 

and then this list is more like
<Box>
<Banner whateverprops={true}/>
<List />
<BetaNote />
```

**Comment by keybase-ci-visdiff** [comment]
:no_entry: Error rendering the commits 65acdd0dbdd8...22531bb5d467 on linux. <details><summary>:mag_right: 4 errored</summary>

 * Error in 65acdd0dbdd8: **Chat_Info_Panel-Normal-ERROR**  
   Chat_Info_Panel-Normal-ERROR rendered
 * Error in 22531bb5d467: **Chat_Info_Panel-Normal-ERROR**  
   Chat_Info_Panel-Normal-ERROR rendered
 * Error in 65acdd0dbdd8: **Chat_Info_Panel-Muted-ERROR**  
   Chat_Info_Panel-Muted-ERROR rendered
 * Error in 22531bb5d467: **Chat_Info_Panel-Muted-ERROR**  
   Chat_Info_Panel-Muted-ERROR rendered
</summary>

**Comment by keybase-ci-visdiff** [comment]
:no_entry: Error rendering the commits 5cab7e43c582...d2dc6a135bcf on linux. <details><summary>:mag_right: 4 errored, 2 changed</summary>

 * Error in 5cab7e43c582: **Chat_Info_Panel-Normal-ERROR**  
   Chat_Info_Panel-Normal-ERROR rendered
 * Error in d2dc6a135bcf: **Chat_Info_Panel-Normal-ERROR**  
   Chat_Info_Panel-Normal-ERROR rendered
 * Error in 5cab7e43c582: **Chat_Info_Panel-Muted-ERROR**  
   Chat_Info_Panel-Muted-ERROR rendered
 * Error in d2dc6a135bcf: **Chat_Info_Panel-Muted-ERROR**  
   Chat_Info_Panel-Muted-ERROR rendered
 * Changed: **Chat_Inbox-You_Rekey** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1504206878724-5cab7e43c582-d2dc6a135bcf-linux/Chat_Inbox-You_Rekey-5cab7e43c582.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1504206878724-5cab7e43c582-d2dc6a135bcf-linux/Chat_Inbox-You_Rekey-d2dc6a135bcf.png)  
   Chat_Inbox-You_Rekey comparison
 * Changed: **Chat_Inbox-Part_Rekey_Selected** [(before)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1504206878724-5cab7e43c582-d2dc6a135bcf-linux/Chat_Inbox-Part_Rekey_Selected-5cab7e43c582.png) [(after)](https://keybase-jenkins-visdiff.s3.amazonaws.com/1504206878724-5cab7e43c582-d2dc6a135bcf-linux/Chat_Inbox-Part_Rekey_Selected-d2dc6a135bcf.png)  
   Chat_Inbox-Part_Rekey_Selected comparison
</summary>

File: shared/teams/render.js.flow (hunk 1 above)
Comment (akalin-keybase) [reply shared/teams/render.js.flow]: Done, added button (behind a flag) on mobile and tweaked stuff to make it look ok on mobile.

File: shared/teams/index.js (hunk 2 above)
Comment (akalin-keybase) [reply shared/teams/index.js]: Done.

File: shared/app/tab-bar/index.render.desktop.js (hunk 3 above)
Comment (akalin-keybase) [reply shared/app/tab-bar/index.render.desktop.js]: Left as is, since it turned out they don't exist yet.

File: shared/teams/render.js (hunk 4 above)
Comment (akalin-keybase) [reply shared/teams/render.js]: Done.

File: shared/teams/render.js (hunk 5 above)
Comment (akalin-keybase) [reply shared/teams/render.js]: Done. Although I left the heights widths etc. as constants.

File: shared/teams/render.js (hunk 7 above)
Comment (akalin-keybase) [reply shared/teams/render.js]: Done.

File: shared/teams/render.js (hunk 6 above)
Comment (akalin-keybase) [reply shared/teams/render.js]: Not in this PR; added TODO.

**Comment by akalin-keybase** [comment]
All comments addressed, PTAL.

File: shared/teams/index.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  onJoinTeam: () => {
+    // TODO: Hook this up once we have a join team dialog.
+    console.log('onJoinTeam not implemented yet')
+  },
+  onReadDoc: () => {
+    openURL('https://keybase.io/docs/command_line/teams_alpha')
+  },
+})
+
+const mergeProps = (stateProps, dispatchProps: DispatchProps) => ({
```
Comment (chrisnojima) [inline shared/teams/index.js]: we dont need the mergeProps if its just going to basically do the default, i'd just remove it

File: shared/teams/render.js.flow [hunk 9]
```diff
@@ -0,0 +1,10 @@
...
+import * as React from 'react'
+
+export type Props = {
```
Comment (chrisnojima) [inline shared/teams/render.js.flow]: we don't need the flow file if there's a unified js file for desktop and native

File: shared/teams/index.js (hunk 8 above)
Comment (akalin-keybase) [reply shared/teams/index.js]: Done. Turns out what I wanted was just to type the return value of `mapDispatchToProps`.

File: shared/teams/render.js.flow (hunk 9 above)
Comment (akalin-keybase) [reply shared/teams/render.js.flow]: Done.

**Comment by akalin-keybase** [comment]
All comments addressed, PTAL!

File: shared/teams/index.js [hunk 10]
```diff
@@ -3,7 +3,11 @@ import Render from './render'
 import pausableConnect from '../util/pausable-connect'
 import openURL from '../util/open-url'
 
-const mapStateToProps = state => ({})
+import type {TypedState} from '../constants/reducer'
+
+type StateProps = {}
+
+const mapStateToProps = (state: TypedState): StateProps => ({})
```
Comment (chrisnojima) [inline shared/teams/index.js]: if you don't actually map any state to props you can just pass null to connect

Comment (akalin-keybase) [reply shared/teams/index.js]: Done.

**Comment by akalin-keybase** [comment]
Merging after CI.

**Comment by chrisnojima** [comment]
👍

**Pull request merged**